### PR TITLE
Migrate codec tests to JUnit 5

### PR DESCRIPTION
**Pull request by kashike**
https://github.com/netty/netty/issues/10757#issuecomment-848445551

- [x] `io.netty.handler.codec.marshalling.AbstractMarshallingTest`

### CODE REVIEW

**Comment by normanmaurer** [comment]
@kashike let me know once this is ready

**Comment by kashike** [comment]
@normanmaurer This should be good to go now.

File: codec/src/test/java/io/netty/handler/codec/marshalling/AbstractMarshallingTest.java [hunk 1]
```diff
@@ -37,6 +38,6 @@ public static void checkSupported() throws Throwable {
             }
             error = cause;
         }
-        Assume.assumeNoException(error);
+        assumeTrue(error == null, error + " was not null");
```
Comment (kashike) [inline codec/src/test/java/io/netty/handler/codec/marshalling/AbstractMarshallingTest.java:41]: There is no replacement for `assumeNoException`, so I *assumed* (ha!) this should work - a second look might be good here.

Comment (normanmaurer) [reply codec/src/test/java/io/netty/handler/codec/marshalling/AbstractMarshallingTest.java:41]: I think this is fine :)

**Comment by normanmaurer** [comment]
@kashike thanks a lot !

**Pull request merged**